simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/handlers/ocpp/CancelReservationRequestHandler.java

```diff
@@ -0,0 +1,58 @@
+package com.evbox.everon.ocpp.simulator.station.handlers.ocpp;
+
+import com.evbox.everon.ocpp.simulator.station.StationMessageSender;
+import com.evbox.everon.ocpp.simulator.station.StationStore;
+import com.evbox.everon.ocpp.simulator.station.evse.Connector;
+import com.evbox.everon.ocpp.v20.message.station.CancelReservationRequest;
+import com.evbox.everon.ocpp.v20.message.station.CancelReservationResponse;
+import com.evbox.everon.ocpp.v20.message.station.Reservation;
+import com.evbox.everon.ocpp.v20.message.station.StatusNotificationRequest;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+public class CancelReservationRequestHandler implements OcppRequestHandler<CancelReservationRequest> {
+
+    private final StationMessageSender stationMessageSender;
+    private final StationStore stationStore;
+
+    @Override
+    public void handle(String callId, CancelReservationRequest request) {
+        Optional<Reservation> reservationOpt = stationStore.tryFindReservationById(request.getReservationId());
+
+        if (!reservationOpt.isPresent()) {
```

Comment:
this if / else block is bad
- it  takes too much responsibilities 
- mixes how and what 
- introduces optional abuse 


it should be restructured in a way 

```
(Optional) reservation = stationStore.tryFindReservationById(request.getReservationId())
calResults = reservation.map(...ACCEPTED).orElseGet(...REJECTED)

reservation.ifPresent(
 - sendNotification
  - removeReservation
)
sendCall(calResults)

```

Reply:
I haven't looked through all the details, however I hope the idea is clear

Reply:
Agreed ... start with the condition in the else block, that is to say if the reservation is present

Reply:
Agreed with Oleg, start the conditional with the case where the reservation is present and use an else block for the case where it is absent

Reply:
Updated